File: alotta-core/src/main/java/liberty/capstone/core/userrestaurant/UserRestaurant.java
package liberty.capstone.core.userrestaurant;

import liberty.capstone.core.appuser.AppUser;
import liberty.capstone.core.restaurant.Restaurant;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class UserRestaurant {
    private AppUser appUser;
    private Restaurant restaurant;
}

File: alotta_client/lib/assets/data/user_restaurant.dart
import 'package:alotta_client/assets/data/app_user.dart';
import 'package:alotta_client/assets/data/restaurant.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_restaurant.g.dart';

@JsonSerializable()
class UserRestaurant {
  final Restaurant restaurant;
  final AppUser appUser;

  const UserRestaurant({
    this.restaurant = const Restaurant(),
    this.appUser = const AppUser(),
  });

  factory UserRestaurant.fromJson(Map<String, dynamic> json) =>
      _$UserRestaurantFromJson(json);

  Map<String, dynamic> toJson() => _$UserRestaurantToJson(this);
}
